### PR TITLE
Add Youtube wall popup to v10

### PR DESCRIPTION
**Pull request by smalluban**
* Moves the UI for the Youtube wall popup into the `packages/ui`
* Unify logic for v8 and v10
* Adds a condition for `showIframe` to detect if the iframe is already shown
* Adds `detectWall` function
* Hides popup when "Open YouTube in a private window" is clicked

The current logic detects changes in the DOM and checks if the URL has changed. If so, selectors for the wall are checked if exists, and then we trigger showing up the iframe. As the iframe blocks displaying more than once, it triggers each time the selectors are found.

For the notice, Safari `chrome.windows.create()` silently returns a resolved promise if the extension is not allowed in private mode. The `chrome.extension.inIncognitoContext` also always returns `false`.

### CODE REVIEW

**Comment by smalluban** [comment]
Initial @kalinme testing failed.

File: extension-manifest-v3/src/content_scripts/youtube.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+// INFO: Safari always returns false for `inIncognitoContext`
+if (!chrome.extension.inIncognitoContext) {
+  (async () => {
+    if (await isFeatureDisabled()) return;
+
+    window.addEventListener('yt-navigate-start', () => closeIframe(), true);
+
+    detectWall(async () => {
+      if (await isFeatureDisabled()) return;
```
Comment (smalluban) [inline extension-manifest-v3/src/content_scripts/youtube.js:43]: YT page is long living app, so the condition must be checked each time - not only at startup (applies to v8 & v10).

**Review by philipp-classen — APPROVED** [review]

**Review by chrmod — APPROVED** [review]
Looks good!

**Review by Bankrollshow — COMMENTED** [review]
- [ ]

File: extension-manifest-v3/src/background/helpers.js [hunk 2]
```diff
@@ -16,6 +16,12 @@ chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
...
       return true;
+    case 'openTabWithUrl':
+      chrome.tabs.create({ url: msg.url });
+      break;
+    case 'openPrivateWindowWithUrl':
+      chrome.windows.create({ url: msg.url, incognito: true });
+      break;
   }
 
   return false;
```
Comment (Bankrollshow) [inline extension-manifest-v3/src/background/helpers.js:27]: - [ ]

File: extension-manifest-v2/app/youtube/index.js [hunk 3]
```diff
@@ -8,10 +8,10 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0
  */
+
```
Comment (Bankrollshow) [inline extension-manifest-v2/app/youtube/index.js:11]: -

**Review by Bankrollshow — COMMENTED** [review]
- [ ]

**Pull request merged**